src/libstd/sys/unix/process/process_unix.rs

```diff
@@ -103,13 +106,8 @@ impl Command {
             }
         }
 
-        fn combine(arr: &[u8]) -> i32 {
-            let a = arr[0] as u32;
-            let b = arr[1] as u32;
-            let c = arr[2] as u32;
-            let d = arr[3] as u32;
-
-            ((a << 24) | (b << 16) | (c << 8) | (d << 0)) as i32
+        fn combine(arr: [u8; 4]) -> i32 {
+            i32::from_be_bytes(arr)
```

Comment:
Same as above: before it always swapped, now it only swaps on little endian system.

Reply:
The generated asms are the same: https://rust.godbolt.org/z/ezbn7K .
IIUC, the old code is loading Big Endian bytes from `arr`. 
arr[0] is most significant value, arr[3] is the least significant one.